packages/libs/src/sdk/api/challenges/ChallengesApi.ts

```diff
@@ -0,0 +1,324 @@
+import { wAUDIO } from '@audius/fixed-decimal'
+import type {
+  ClaimableTokensClient,
+  DiscoveryNodeSelectorService,
+  LoggerService
+} from '../../services'
+import type {
+  AntiAbuseOracle,
+  AntiAbuseOracleSelectorService
+} from '../../services/AntiAbuseOracleSelector/types'
+import type { RewardManagerClient } from '../../services/Solana/programs/RewardManagerClient/RewardManagerClient'
+import { parseParams } from '../../utils/parseParams'
+import { BaseAPI, Configuration } from '../generated/default'
+import {
+  ChallengesApi as GeneratedChallengesApi,
+  Configuration as ConfigurationFull
+} from '../generated/full'
+import type { UsersApi } from '../users/UsersApi'
+import {
+  ChallengeId,
+  ClaimRewardsRequest,
+  ClaimRewardsSchema,
+  GenerateSpecifierRequest,
+  GenerateSpecifierSchema
+} from './types'
+import type { PublicKey } from '@solana/web3.js'
+import { AntiAbuseOracleApi } from '../antiAbuseOracle/AntiAbuseOracleApi'
+import { toChecksumAddress } from 'ethereumjs-util'
+
+export class ChallengesApi extends BaseAPI {
+  private readonly logger: LoggerService
+  constructor(
+    config: Configuration,
+    private readonly usersApi: UsersApi,
+    private readonly discoveryNodeSelector: DiscoveryNodeSelectorService,
+    private readonly rewardManager: RewardManagerClient,
+    private readonly claimableTokens: ClaimableTokensClient,
+    private readonly antiAbuseOracleSelector: AntiAbuseOracleSelectorService,
+    logger: LoggerService
+  ) {
+    super(config)
+    this.logger = logger.createPrefixedLogger('[challenges-api]')
+  }
+
+  /**
+   * Formats the specifier for a claimable instance of a challenge.
+   *
+   * For most challenges, this is the user ID of the user that's claiming the
+   * the challenge reward. For challenges like referrals or $AUDIO matching,
+   * this includes more data like the referred user ID or purchased track ID
+   * to allow the challenge to be claimed multiple times, but only once per
+   * completed instance.
+   *
+   * @hidden
+   */
+  public async generateSpecifier(request: GenerateSpecifierRequest) {
```

Comment:
I don't fully follow the bit about disallowing specifiers.
But generally are you saying it's odd that you have to discover the values for the specifier somehow and construct this object in order to claim a challenge, when it would be just as feasible to have the SDK tell you you have x AUDIO available and then you just claim(x AUDIO) and it works out the right values for you?

Reply:
"disallowing" was probably the wrong word, but basically if we abstract it further how would we support the existing flow which uses specifiers _and_ support the new flow which generates them on the fly with the proper parameters? I think this is a small issue but wanted to call it out as it's sorta against the principle of keeping the external interface extremely approachable for third parties ("what is a 'specifier'?")